sklearn/utils/tests/test_extmath.py

```diff
@@ -275,6 +271,44 @@ def test_randomized_eigsh_compared_to_others(k):
         assert_array_almost_equal(eigvecs_arpack, eigvecs_lapack, decimal=8)
 
 
+@pytest.mark.parametrize("k", (10, 50, 100, 199, 200))
+def test_randomized_eigsh_value_compared_to_others(k):
+    """Check that `_randomized_eigsh(value)` is similar to other `eigsh`
+
+    Tests that for a random PSD matrix, `_randomized_eigsh(value)` provides results
+    comparable to LAPACK (scipy.linalg.eigh)
+    """
+    n_features = 200
+    # make a random PSD matrix
+    X = make_sparse_spd_matrix(n_features, random_state=0)
```

Comment:
This is PSD, but we should also check non-PSD. So let's create a symmetric random matrix instead. If you wish you can test the two cases

- add a `@pytest.mark.parametrize("is_psd", (True, False))`
- in the test, switch on `if is_psd: ` to create the random X.